Src/VimCore/CommonOperations.fs

```diff
@@ -2152,6 +2209,14 @@ type internal CommonOperations
             x.AdjustTextViewForScrollOffsetAtPoint point
         if Util.IsFlagSet viewFlags ViewFlags.VirtualEdit && point.Position = x.CaretPoint.Position then
             x.AdjustCaretForVirtualEdit()
+    
+    /// Ensure the view properties are met at the caret
+    member x.EnsureAtCaret viewFlags = 
+        x.DoActionWhenReady (fun () -> x.EnsureAtPointSync x.CaretPoint viewFlags)

```

Comment:
During the initial load is it possible for this method to get invoked multiple times hence resulting in it executing multiple times once the underlying `ITextView` is ready? These functions are idempotent in isolation hence so I don't think this is a problem even if it does happen. But wanted to make sure you agree on this.

Reply:
Yes, we already had the issue of it possibly running multiple times, but those multiple times would be synchronous, not posted. The `EnsureAtCaret` is indeed idempotent as long as the buffer is unmodified and the caret isn't moved. I believe that posted actions are stable in that if action A is posted before action B, A will be scheduled before B.

However, since we don't **always** post, it is possible for us to post an `EnsureAtCaret` and a synchronous call to `EnsureAtCaret` to run first. However, in that case I believe that is just the order we would want them to run in. In either case, it is not non-deterministic.